File: chess-game/chess-game.v3/core/domain/pieces/piece_factory.py
from core.domain.pieces.bishop import Bishop
from core.domain.pieces.king import King
from core.domain.pieces.knight import Knight
from core.domain.pieces.pawn import Pawn
from core.domain.pieces.piece_type import PieceType
from core.domain.pieces.queen import Queen
from core.domain.pieces.rook import Rook
from core.domain.value_objects.piece_id import PieceId
from core.domain.value_objects.side import Side


class PieceFactory(object):

    @classmethod
    def create(cls, piece_id: PieceId, side: Side, type_: PieceType):
        match type_:
            case PieceType.Pawn:
                return Pawn(piece_id, side)
            case PieceType.King:
                return King(piece_id, side)
            case PieceType.Rook:
                return Rook(piece_id, side)
            case PieceType.Queen:
                return Queen(piece_id, side)
            case PieceType.Bishop:
                return Bishop(piece_id, side)
            case PieceType.Knight:
                return Knight(piece_id, side)
        return None

File: chess-game/chess-game.v3/chessapp/interface/client/chess-app/src/app/pages/game/play/play/models/pieces/piece_factory.ts
import { PieceDto } from "src/app/services/models/chess-game-dto";
import { Side } from "../side";
import { Bishop } from "./bishop";
import { King } from "./king";
import { Knight } from "./knight";
import { Pawn } from "./pawn";
import { Queen } from "./queen";
import { Rook } from "./rook";

export class PieceFactory {
    
    static getPiece(piece: PieceDto) {
        switch (piece.abbreviation.toLowerCase()) {
            case 'wp':
                return new Pawn(piece.piece_id, Side.white);
            case 'bp':
                return new Pawn(piece.piece_id, Side.black);
            case 'wr':
                return new Rook(piece.piece_id, Side.white);
            case 'br':
                return new Rook(piece.piece_id, Side.black);
            case 'wb':
                return new Bishop(piece.piece_id, Side.white);
            case 'bb':
                return new Bishop(piece.piece_id, Side.black);
            case 'wn':
                return new Knight(piece.piece_id, Side.white);
            case 'bn':
                return new Knight(piece.piece_id, Side.black);
            case 'wq':
                return new Queen(piece.piece_id, Side.white);
            case 'bq':
                return new Queen(piece.piece_id, Side.black);
            case 'wk':
                return new King(piece.piece_id, Side.white);
            case 'bk':
                return new King(piece.piece_id, Side.black);
            default:
                throw Error('Un-recognized piece type');
        }
    }
}